### PR TITLE
Support modular explorer

### PR DESCRIPTION
**Pull request by mengweieric**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,10 +81,6 @@ Observability is collection of plugins and applications that let you visualize d
 [bug-link]: https://github.com/opensearch-project/observability/issues?q=is%3Aopen+is%3Aissue+label%3Abug+
 [enhancement-badge]: https://img.shields.io/github/issues/opensearch-project/observability/enhancement.svg
 [enhancement-link]: https://github.com/opensearch-project/observability/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement+
-<<<<<<< HEAD
```
Comment (eugenesk24) [inline README.md:84]: We should probably remove these conflict flags

File: dashboards-observability/public/components/common/search/search.tsx [hunk 2]
```diff
@@ -83,20 +85,22 @@ export const Search = (props: any) => {
...
-          return !staleState;
-        });
-      }}
-      data-test-subj="eventExplorer__saveManagementPopover"
-      iconType="arrowDown"
-    >
-      Save
-    </EuiButton>
-  );
+  const Savebutton = useMemo(() => {
```
Comment (eugenesk24) [inline dashboards-observability/public/components/common/search/search.tsx:88]: Why do we need to useMemo here?? It doesn't seem like `searchBarConfigs` or `selectedSubTabId` affect the save button?

Comment (mengweieric) [reply dashboards-observability/public/components/common/search/search.tsx:88]: yea, previously was used to toggle disable flag...missed this

File: README.md (hunk 1 above)
Comment (mengweieric) [reply README.md:84]: These changes are made to remove the conflicts...

Comment (eugenesk24) [reply README.md:84]: Ah. yes you are correct.

**Review by eugenesk24 — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**